generator/lib/src/realm_model_info.dart

```diff
@@ -48,10 +48,13 @@ class RealmModelInfo {
         yield* required.map((f) => '${f.mappedTypeName} ${f.name},');
 
         final notRequired = allSettable.where((f) => !f.isRequired && !f.isPrimaryKey);
-        final collections = fields.where((f) => f.type.isRealmCollection).toList();
+        final collections = fields.where((f) => f.isRealmCollection).toList();
         if (notRequired.isNotEmpty || collections.isNotEmpty) {
           yield '{';
-          yield* notRequired.map((f) => '${f.mappedTypeName} ${f.name}${f.hasDefaultValue ? ' = ${f.fieldElement.initializerExpression}' : ''},');
+          yield* notRequired.map(
+            (f) =>
+                '${f.mappedTypeName} ${f.name}${f.hasDefaultValue ? ' = ${f.fieldElement.initializerExpression}' : (f.isMixed ? ' = const RealmValue.nullValue()' : '')},',
```

Comment:
That's kinda convoluted. Can we make a few adjustments to `RealmFieldInfo` to accommodate mixed support? E.g. we can change things to:

```dart
class RealmFieldInfo {
  // ...
  bool get hasDefaultValue => fieldElement.hasInitializer || isMixed;

  String? get defaultValueExpression {
    if (fieldElement.hasInitializer) {
      return fieldElement.initializerExpression.toString();
    }

    if (isMixed) {
      return 'const RealmValue.nullValue()';
    }

    return null;
  }
}
```

Then this will simplify to:

```dart
yield* notRequired.map((f) => '${f.mappedTypeName} ${f.name}${f.hasDefaultValue ? ' = ${f.defaultValueExpression}' : ''},');
```

Reply:
I will rework this to make intent more clear

Reply:
Simplified